core/inmem_worker.go

```diff
@@ -0,0 +1,419 @@
+package core
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"log"
+	"os/exec"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type streamingOutputRequest struct {
+	id         StreamingOutputRequestID
+	jobID      JobID
+	outChannel chan []byte
+}
+
+type inMemWorker struct {
+	newJobRequestQueue  chan *Job
+	stopJobRequestQueue chan JobID
+
+	jobs              map[JobID]*Job
+	jobsMapLock       *sync.RWMutex
```

Comment:
Nit: None of these locks need to be pointers (same throughout).
```suggestion
	jobsMapLock       sync.RWMutex
```

Reply:
Suggestion has been applied